apps/evm/lib/evm/operation/system.ex

```diff
@@ -37,22 +26,27 @@ defmodule EVM.Operation.System do
       if is_allowed do
         available_gas = Helpers.all_but_one_64th(machine_state.gas)
 
+        {account_interface, _nonce} =
+          AccountInterface.increment_account_nonce(exec_env.account_interface, exec_env.address)
```

Comment:
The actual fix, see the screenshot in the PR description.
I think this is correct, just need someone to confirm that I'm right